### DISCUSSION TITLE
setFocus unhelpful error message

### DISCUSSION THREAD

**Discussion by JoaoSilveira** [react-hook-form/react-hook-form#5165] [Q&A]
**Describe the bug**

I get an unhelpful message when calling `setFocus` for a field that does not set its `ref` prop on **React Native**. Error message:
`ERROR  TypeError: undefined is not a function, js engine: hermes`

**To Reproduce**

1. Use React Native
2. Create a `<Controller>`
2. Do not set `ref={ref}` on the input component
3. Call `setFocus` for the field

My code was something like the following. (did not test this one though)

```javascript
function Component() {
  const form = useForm();

  setEffect(() => form.setFocus('field'), [form.setFocus]);

  return (
    <Controller
      name="field"
      control={form.control}
      render={({ field: { value, onChange, ref } }) => (
        <TextInput
          placeholder="The Field"
          value={value}
          onChangeText={onChange}
          // note that it is missing ref={ref}
          // if I add it everything work as expected
        />
      )}
    />
  );
}
```

**Codesandbox link**

I reproduced the issue in React (not native). Here the message is: `get(...)._f.ref.focus is not a function`

https://codesandbox.io/s/relaxed-leavitt-npn0f

**Expected behavior**
I'd like to receive a more helpful error, something like:
`ERROR: attempt to set focus on input named [name] that does not support it. Is the input perhaps missing ref={ref}?`

**Smartphone:**

- Emulator: Android emulator version 30.5.3.0 (build_id 7196367)
- OS: Android 11
- React Version: 17.0.1
- React Native Version: 0.64.0
- React Hook Form Version: 7.5.2

**Comment by bluebill1049** [comment]
I will update the doc, this API is only applicable for the web.

**Reply by bluebill1049** [reply]
<img width="772" alt="Screen Shot 2021-05-12 at 8 55 45 am" src="https://user-images.githubusercontent.com/10513364/117894539-e26b0080-b2ff-11eb-9912-82ad53101386.png">

doc is updated, the lib invokes `focus` method. react-native doesn't have such a method.

**Reply by JoaoSilveira** [reply]
React Native [does have `focus`](https://reactnative.dev/docs/direct-manipulation#focus), though it is limited to [focusable Views](https://reactnative.dev/docs/direct-manipulation#focus) and [TextInputs](https://reactnative.dev/docs/textinput#focus) (perhaps [`TouchableWithoutFeedback`](https://reactnative.dev/docs/touchablewithoutfeedback#onfocus) also does have it because it has `onFocus` event)

**Reply by bluebill1049** [reply]
can we store that inside the `ref`? otherwise, hook form won't be able to reach it.

**Reply by JoaoSilveira** [reply]
Yes, it can be done. In fact, the `setFocus` does work when I add `ref={ref}` to the `TextInput`

**Reply by bluebill1049** [reply]
ok, i will update the doc on it.

**Reply by JoaoSilveira** [reply]
I just wanted a better error message for trying to `setFocus` on an "unregistered" component (that is, a component that does not set the `ref`), because you can't `{...register('field')}` on Native
